spec/controllers/v1/people_controller_spec.rb

```diff
@@ -14,4 +13,75 @@
     end
   end
 
+  describe "GET targets" do
+    context "no params" do
+      before { get :targets }
+
+      it "returns error" do
+        expect(assigns(:error)).to_not be_blank
+      end
+    end
+
+    context "bad email" do
+      before { get :targets, { email: 'bad' } }
+
+      it "returns error" do
+        expect(assigns(:error)).to_not be_blank
+      end
+    end
+
+    context "good email" do
+      context "no address" do
+        context "new user" do
+
+          it "returns address required" do
+            get :targets, { email: 'user@example.com' }
+            expect(assigns(:address_required)).to be true
+          end
+
+          it "doesn't update location" do
+            expect_any_instance_of(Location).to receive(:update_location).
+              with(address: nil, city: nil, state: nil, zip: nil) { nil }
+
+            get :targets, { email: 'user@example.com' }
+          end
+
+          it "sets target legislators" do
+            rep = instance_double("Legislator")
+            expect_any_instance_of(Person).to receive(:target_legislators).
+              with(json: true) { [rep] }
+            get :targets, { email: 'user@example.com' }
+            expect(assigns(:target_legislators)).to eq [rep]
+          end
+
+        end
+
+        context "existing user with address info" do
+
+          before do
+            FactoryGirl.create(:person, email: 'user@example.com',
+              district: FactoryGirl.create(:district))
+            get :targets, { email: 'user@example.com' }
+          end
+
+          it "returns address not required" do
+            expect(assigns(:address_required)).to be false
+          end
+
+        end
+      end
+
+      context "good address" do
+
+        it "updates location" do
+          expect_any_instance_of(Location).to receive(:update_location).
+            with(address: '2020 Oregon St', city: nil, state: nil, zip: '94703') { true }
```

Comment:
Line is too long. [89/80]<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.